proto/redpanda/api/dataplane/v1alpha1/user.proto

```diff
@@ -0,0 +1,196 @@
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128
+  ];
+}
+
+message DeleteUserResponse {}
+
+service UserService {
+  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {
+    option (google.api.http) = {
+      post: "/v1alpha1/users"
+      body: "user"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Create User"
+      description: "Create a User."
+      responses: {
+        key: "201"
+        value: {
+          description: "Creates a user"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.CreateUserRequest.User"}
+          }
+        }
+      }
+      responses: {
+        key: "400"
+        value: {
+          description: "Bad Request. Check API documentation and update request."
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {
+    option (google.api.http) = {
+      put: "/v1alpha1/users/{user.name}"
+      body: "user"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Update User"
+      description: "Update a User."
+      responses: {
+        key: "200"
+        value: {
+          description: "Ok"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.UpdateUserResponse.User"}
+          }
+        }
+      }
+      responses: {
+        key: "400"
+        value: {
+          description: "Bad Request. Check API documentation and update request."
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+  rpc ListUsers(ListUsersRequest) returns (ListUsersResponse) {
+    option (google.api.http) = {get: "/v1alpha1/users"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List Users"
+      description: "List Users"
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.ListUsersResponse"}
+          }
+        }
+      }
+    };
+  }
+  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {
+    option (google.api.http) = {delete: "/v1alpha1/users/{name}"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Delete Users"
+      description: "Delete Users"
+      responses: {
+        key: "204"
+        value: {
+          description: "User was deleted successfully."
+          schema: {}
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+}
```

Comment:
Should this file have comments?

Reply:
What kind of comments should we add here?

Reply:
Just API comments for the types and calls so that it gets generated into the open api v3 spec for API users to have.

Reply:
We have some additional editing of the generated openapi spec (which considers the grpc gateway annotations), in the backend/tools/openapi-generator/main.go . Or you can check the openapi-spec.json to see the final result. I think this already covers examples and short descriptions.